tests/test_kernels.py

```diff
@@ -0,0 +1,235 @@
+import parcels.application_kernels
+import plasticparcels as pp
+import parcels
+import numpy as np
+from datetime import datetime, timedelta
+import pytest
+
+
+def make_standard_simulation_settings():
+    simulation_settings = {'startdate': datetime.strptime('2020-01-04-00:00:00', '%Y-%m-%d-%H:%M:%S'),
+                           'runtime': timedelta(days=2),
+                           'outputdt': timedelta(hours=1),
+                           'dt': timedelta(minutes=20),
+                           }
+    return simulation_settings
+
+
+def make_standard_plastictype_settings():
+    # Use tiny wind percentage because test data set is not large and wind speeds are quick!
+    plastictype_settings = {'wind_coefficient': 0.0001,     # Percentage of wind to apply to particles
+                            'plastic_diameter': 0.001,  # Plastic particle diameter (m)
+                            'plastic_density': 1030.,   # Plastic particle density (kg/m^3)
+                            }
+    return plastictype_settings
+
+
+def make_standard_particleset(fieldset, settings):
+    # Generate a particleset that has particles in the test domain
+    settings['release'] = {'initialisation_type': 'fisheries', 'country': 'Malta'}
+    pset = pp.constructors.create_particleset_from_map(fieldset, settings)
+
+    # Only keep particles in the test domain
+    keep_particles = (pset.lon > 17) & (pset.lon < 20) & (pset.lat < 36) & (pset.lat > 34)
+    pset.remove_booleanvector(~keep_particles)
+
+    return pset
+
+
+@pytest.mark.parametrize('use_3D', [True, False])
+def test_advection_only(use_3D):
+    settings_file = 'tests/test_data/test_settings.json'
+    settings = pp.utils.load_settings(settings_file)
+    settings = pp.utils.download_plasticparcels_dataset('NEMO0083', settings, 'input_data')
```

Comment:
Why is this call to `download_plasticparcels_dataset` needed, if the NEMO-data are not used? See also in the other examples

Reply:
In some of the test functions this can be removed, but in the cases where Stokes/Wind is used (and `kernels = pp.constructors.create_kernel(...)`), then the unbeaching kernel is included in that list. I'll remove this call to `download_plasticparcels_dataset` for the cases where the unbeaching kernel is not invoked.

Reply:
Actually, I've just realised, we use it in the `make_standard_particleset` call, which creates the particleset used to test the advection.
I thought this would be easier, since the particleset requires a number of variables used by the different kernels. So I can't remove this from each of the tests without writing a new function for particleset creation.

Reply:
Hmm, in the spirit of modularisation it may then be better to split it up into different functions at some point? Perhaps in a later PR?

Reply:
I agree, a class for `PlasticParticle` would be a cleaner approach, so that users can still use the `parcels.ParticleSet.from_list()` approach, with their own release locations.